lib/src/endpoints/users.dart

```diff
@@ -18,11 +18,23 @@ class Users extends EndpointPaging {
 
   Future<Player> currentlyPlaying() async {
     var jsonString = await _api._get('v1/me/player/currently-playing');
-    var map = json.decode(jsonString);
 
+    if (jsonString.isEmpty) {
```

Comment:
so we want this check here because it seems like if you're not currently-playing something then the jsonString is empty. Seems to make more sense to return a Player instead of an error being raise on json.decode(). The player object also states in the documentation that the track could be null, so this lines up with returning a Player that isn't playing anything

Reply:
"When no track is currently playing, the request will return a 204 NO CONTENT response with no payload. If private session is enabled the response will be a 204 NO CONTENT with an empty payload." - https://developer.spotify.com/documentation/web-api/reference/player/get-the-users-currently-playing-track/

Reply:
Good catch, I think that makes sense of a library consumer standpoint!